### PR TITLE
Update language around assumptions around files and directories

### LINKED ISSUE #495: assumptions around modifying input files, or making new files in their directories
It is suggested the the OpenWDL specification clarify the write-status of input files, and the directories they are in.

I suggest that in a future version of the OpenWDL spec, it is declared that all inputs files must be make read-only to have consistent behavior.

This also helps with converting to CWL, as it has the same restriction, unless `InitialWorkDirRequirement` is used to mark some inputs as `writable: true`

For miniwdl, they have an IO-expensive workaround 
https://miniwdl.readthedocs.io/en/latest/runner_advanced.html#read-only-input-files

### PR DESCRIPTION
**Pull request by vsmalladi**
### Checklist
- [x] Pull request details were added to CHANGELOG.md
- [x] Valid examples WDL's were added or updated to the SPEC.md (see the [guide](https://github.com/openwdl/wdl-tests/blob/main/docs/MarkdownTests.md) on writing markdown tests)

Closes #495

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -576,6 +578,8 @@ The following primitive types exist in WDL:
       Float f = 27.3
       String s = "hello, world"
       File x = write_file_task.x
+    ##TODO
```
Comment (rhpvorderman) [inline SPEC.md]: I see a todo left in here.

**Review by rhpvorderman — APPROVED** [review]
Looks good to me, but I found a leftover `TODO` comment.

File: SPEC.md (hunk 1 above)
Comment (vsmalladi) [reply SPEC.md]: Fixed

**Comment by vsmalladi** [comment]
@rhpvorderman can you re-review?

**Review by patmagee — APPROVED** [review]
LGTM

**Pull request merged**